spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -140,6 +140,20 @@ sealed class SpdxExpression {
     fun isValid(strictness: Strictness = Strictness.ALLOW_CURRENT): Boolean =
         runCatching { validate(strictness) }.isSuccess
 
+    /**
+     * Return true if [choice] is a valid license choice for this SPDX expression. This is the case, if [choice] does
+     * not contain the [OR operator][SpdxOperator.OR], and if the [licenses][SpdxSingleLicenseExpression] contained in
+     * [choice] match any of the [valid license choices][validChoices].
+     */
+    fun isValidChoice(choice: SpdxExpression): Boolean {
```

Comment:
I've brought this up in our call already, not sure if it became that clear though.
I write it again to bring back the awareness to make a conscious decision:

If we implement license choice based on this function, any flakyness from ScanCode or any false positive detection can render the `isValidChoice` function to return false. Thus we'd be commiting to (1) any flakyness in ScanCode (2) accidental detection of ScanCode (3) any new detection capabilities added to scancode would require the re-work of the choice. It means these three points directly cause human time consumption. I am against doing this but leave the decision up to the rest of the ORT team.

Note: I'm just speculating how this is going to be used (as I do not know it).

Reply:
This is just the basic check if a choice is valid, any more advanced logic, like applying a choice to a subexpression, can be implemented when we add license choice support to ORT.